sklearn/model_selection/_search.py

```diff
@@ -106,11 +107,25 @@ def __init__(self, param_grid):
         for grid in param_grid:
             if not isinstance(grid, dict):
                 raise TypeError("Parameter grid is not a dict ({!r})".format(grid))
-            for key in grid:
-                if not isinstance(grid[key], Iterable):
+            for key, value in grid.items():
+                if isinstance(value, np.ndarray) and value.ndim > 1:
+                    raise ValueError(
+                        f"Parameter array for {key} should be one-dimensional, got:"
+                        f" {value!r} with shape {value.shape}"
+                    )
+                if isinstance(value, str) or not isinstance(
+                    value, (np.ndarray, Sequence)
+                ):
                     raise TypeError(
-                        "Parameter grid value is not iterable "
-                        "(key={!r}, value={!r})".format(key, grid[key])
+                        f"Parameter grid for parameter {key!r} needs to be a list or a"
+                        f" numpy array, but got {value!r} (of type {type(value)})"
```

Comment:
For consistency with the above message:

```suggestion
                        f" numpy array, but got {value!r} (of type {type(value).__name__})"
```

this will require reformatting this file with the `black` command.